src/utils.ts

```diff
@@ -35,8 +35,8 @@ export function loadScript(id: string, config: Pick<VueGtmUseOptions, "defer" |
     return;
   }
 
-  script.async = true;
-  script.defer = config.defer || false;
+  script.async = !config.defer;
+  script.defer = (config.defer || config.compability) as boolean;
```

Comment:
`as boolean` looks wired to me in this place
Maybe we can use `??` operator here to help us to not use a cast
So maybe `config.defer ?? config.compability ?? false` would be better here :thinking:

Reply:
Thing is you apply these default parameters on a different place. If you then here also add false as a fallback you do it twice and you have 2 points of truth

Reply:
It was just to stop ts from screaming, having undefined would also result in a falsy value and would work properly, one could also do it like
```
Boolean(config.defer || config.compability)
```
The standard || should be fine here, if you do not want it like this we can just do || false as it was before. Either if defer or compoability is true, it should evaluate to true, else false.

Reply:
`Boolean(config.defer || config.compability)` is an acceptable solution for me